### ISSUE TITLE
files deleted, but no space is recovered

### ISSUE THREAD

**Issue by batebbateb** [lostindark/DriverStoreExplorer#11]
RAPR finished its process in 2-3 seconds, 14 GB of data freed from the folder, but the free space on the disk has not changed. 
Windows now reports 403 GB of used space.
TreeSize says there is 417 GB of used space.
I have cleaned the Recycle Bin, but I can’t figure out where those 14 GB go.
I have disabled the system recovery, no hibernation is enabled.
Thanks!

**Comment by wonder-wu** [comment]
I have the same problem...
Dont know why.

**Comment by lostindark** [comment]
DriverStore Explorer is using pnputil.exe for deleting drivers:
pnputil.exe /d <infname>

Pnputil.exe is a tool included in Windows. I don't know what the actual issue is.
Have you tried to check the disk for errors?

**Comment by 3dscan** [comment]
Same problem here. Had about 10Gb left on my HD. There were 47 old nvidia drivers taking up about 15Gb of space. After I deleted them I still only see the 10Gb of free space even though the log says all the drivers were deleted succesfully?!
The folders in the File Repository are no where to be found anymore in the explorer and RAPR aswell as pnputil.exe doesn't list the old drivers anymore.
I tried rebooting/recylce bin/disk cleaner/ccleaner but no succes.
Any suggestions are welcome, thnx!

**Comment by lostindark** [comment]
@batebbateb @wonder-wu @3dscan What is the OS version?

**Comment by wonder-wu** [comment]
@lostindark  My OS is windows 10 RS2

**Comment by 3dscan** [comment]
OS is windows 10 64-bit (version 1607), the files were on a Liteonit lcs-256m6s SSD

**Comment by sein-tao** [comment]
Same problem here, on windows 10 Pro (1607). chkdsk reported the same disk usage as in explorer (no space free'd), and no disk problem.

**Comment by lostindark** [comment]
I just tested on my system Version 1703 Build 15063.447. The space was released correctly (in seconds).

**Comment by Ayers-Deng** [comment]
I have the problem too...
OS: Win10 10.0.14393

**Comment by Vinside** [comment]
Did someone find a solution for this?! 

This is crazy 15GB of my SDD can't be used right now because of inproper uninstallation I guess...

**Comment by Vinside** [comment]
I'm on windows 1607 build: 14393.1770, what should I do. Is it because of my windows version ?

**Comment by Ayers-Deng** [comment]
Hi all,

When
  1. I through "Windows Update" update windows version to 1703 Build 15063.608 and
  2. Clean Disk for drive C:\
It seems get back the space.

**Comment by TonyWang8904** [comment]
I have the same problem...
Windows RS1

**Comment by raulqf** [comment]
I have the same problem... The windows size on disk shows the correct size but the size property does not. In fact, I think the size property must be updated but I haven't found yet the way to do it. Any idea?

**Comment by raulqf** [comment]
I have the same problem... The windows size on disk shows the correct size but the size property does not. In fact, I think the size property must be updated but I haven't found yet the way to do it. Any idea?

**Comment by lostindark** [comment]
@TonyWang8904 @raulqf This could be a bug in older version of Windows 10. AFAIK 1703 and 1709 doesn't have this issue. You can try to update to the latest version.

**Comment by raulqf** [comment]
I also upgraded to windows 1709 but still has the same problem. Do you expect that windows update should fix it automatically or may I have to perform any kind of update over windows folder using DriverStore?

**Comment by yufeiying** [comment]
I agree with Riclim. Windows System Restore component keeps the files.

**Comment by raulqf** [comment]
I have windows restore disabled...

**Comment by lostindark** [comment]
@raulqf you can check the size of C:\Windows\System32\DriverStore. If the size changed after you removed driver using the tool, but the disk free space doesn't change. Most likely something else kept the drivers copy (likely System Restore).
Also, check device manager for the device to see if it has "Roll Back Driver" button enabled. If so, that means the driver hasn't been uninstalled.

**Comment by sxq112358** [comment]
Same problem. But I didn't use this DriverStoreExplorer, just use pnputil.exe to delete the drivers. I'm on win 10 1709.

**Comment by zhang99969** [comment]
Same problem. Win10 1803,  windows restore  and "Roll Back Driver" button disabled

**Comment by english-breakfast** [comment]
Same issue here. Anybody found a solution yet? Running Windows Update didn't help. System Restore isn't the culprit.

**Comment by dyyshxm** [comment]
Same problem. Win10 1803. Before cleaning , i checked the nv** folders, about 13G, then after the cleaning, files gone, but the space not back.

**Comment by Here2Tinker** [comment]
Hello!

I just used DriverStore Explorer v0.10.39 on a Windows 10 Home updated, to do a clean-up of about 15 GB of old nvidia drivers. Same problem as you guys... 

The folder C:\Windows\System32\DriverStore\FileRepository got shrunk by that much, but the free space on C drive remained the same.

Has anyone been able to resolve this mystery? I've tried a number of tricks I found in forums, but the issue is still present. 

Any thoughts?

**Comment by lostindark** [comment]
Unfortunately this is an issue which I never be able to repro. Given the files gets deleted correctly and the folder are showing the right size, I assume either the files were kept somewhere else by windows (e.g. system restore), or the windows hasn't update the free space number.

Here is few things to try:
1. Make sure System Restore is not holding files (checking if the feature is turned on or not).
2. Check disk for errors. (chkdsk /f)
3. Optimize/Defrag drive.
4. Monitor the free space for the drive for the next couple of days and see if it changes.

**Comment by Here2Tinker** [comment]
**1-3**  done

**sfc /scannow** showed some errors but mostly about duplicate ownership of some folders
**DISM /Online /Cleanup-Image /CheckHealth** is clean
**DISM /Online /Cleanup-Image /ScanHealth** returned "The component store is repairable." so I ran /RestoreHealth and checked the file system again. No errors. No space gained.

I will post back in a month if system manages to reclaim some space

**Comment by AbnerZhou** [comment]
@batebbateb   The free space is appeared  after i update my os.   Now  my os version is Windows 10 version 1803.

**Comment by Here2Tinker** [comment]
Version 1803 is from April 2018, it'll be end-of-service in November 2019... Which version were you on previously?

**Comment by gbhug5a** [comment]
Has this issue been resolved, at least on newer versions of Windows?  If not, I was thinking SpaceSniffer might show where the deleted drivers have been moved.   You could take screen captures of the SS C drive map immediately before and after deleting the drivers.  That might at least show whether the drivers have been moved somewhere else, or Windows is just reporting the drive free space wrong.
